### PR TITLE
Add PropertyGroup mediator

### PR DESCRIPTION
**Pull request by praveennadarajah**
## Purpose
Adding property-group mediator to EI-Tooling

### CODE REVIEW

File: plugins/org.wso2.developerstudio.eclipse.esb.libraries/.classpath [hunk 1]
```diff
@@ -1,6 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <classpath>
-	<classpathentry exported="true" kind="lib" path="lib/synapse-core_2.1.7.wso2v83.jar"/>
+	<classpathentry exported="true" kind="lib" path="lib/synapse-core_2.1.7.wso2v87.jar"/>
```
Comment (rosensilva) [inline plugins/org.wso2.developerstudio.eclipse.esb.libraries/.classpath]: Shall we use the released synapse-core version?

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/deserializer/PropertyGroupMediatorDeserializer.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+    @Override
+    public PropertyGroupMediator createNode(IGraphicalEditPart part, AbstractMediator mediator)
+            throws DeserializerException {
+        Assert.isTrue(mediator instanceof org.apache.synapse.mediators.builtin.PropertyGroupMediator,
+                "Unsupported mediator passed in for deserialization at " + this.getClass());
+
+        org.apache.synapse.mediators.builtin.PropertyGroupMediator propertyGroupMediator = (org.apache.synapse.mediators.builtin.PropertyGroupMediator) mediator;
+
+        org.wso2.developerstudio.eclipse.gmf.esb.PropertyGroupMediator visualLog = (org.wso2.developerstudio.eclipse.gmf.esb.PropertyGroupMediator) DeserializerUtils
```
Comment (rosensilva) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/deserializer/PropertyGroupMediatorDeserializer.java]: shall we change 'visualLog' to 'visualPropertyGroup'

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/deserializer/PropertyMediatorDeserializer.java [hunk 3]
```diff
@@ -104,6 +104,7 @@ public PropertyMediator createNode(IGraphicalEditPart part, AbstractMediator med
             if (propertyMediator.getExpression() != null) {
                 SynapsePath xpath = propertyMediator.getExpression();
                 NamespacedProperty namespaceProp = createNamespacedProperty(xpath);
+                namespaceProp.setSupportJsonPaths(true);
```
Comment (rosensilva) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/deserializer/PropertyMediatorDeserializer.java]: Are we supporting JSON paths for the property mediator?

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/navigator/EsbNavigatorActionProvider.java [hunk 4]
```diff
@@ -140,7 +140,7 @@ public void run() {
             IEditorInput editorInput = getEditorInput(myDiagram);
             IWorkbenchPage page = myViewerSite.getPage();
             try {
-                page.openEditor(editorInput, EsbDiagramEditor.ID, true, IWorkbenchPage.MATCH_INPUT);
+                page.openEditor(editorInput, EsbDiagramEditor.ID);
```
Comment (rosensilva) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/navigator/EsbNavigatorActionProvider.java]: Why do we need to change this line?

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/navigator/EsbNavigatorContentProvider.java [hunk 5]
```diff
@@ -1908,6 +1908,11 @@ public void init(ICommonContentExtensionSite aConfig) {
             connectedViews = getChildrenByType(connectedViews,
                     EsbVisualIDRegistry.getType(PropertyMediatorEditPart.VISUAL_ID));
             result.addAll(createNavigatorItems(connectedViews, parentElement, false));
+            connectedViews = getChildrenByType(Collections.singleton(sv),
+                    EsbVisualIDRegistry.getType(MediatorFlowMediatorFlowCompartment2EditPart.VISUAL_ID));
```
Comment (rosensilva) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/navigator/EsbNavigatorContentProvider.java]: Formatting issue?

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/navigator/EsbNavigatorContentProvider.java [hunk 6]
```diff
@@ -5846,1007 +5896,749 @@ public void init(ICommonContentExtensionSite aConfig) {
         }
 
         case MediatorFlowEditPart.VISUAL_ID: {
-            return mediatorFlowEditPart(view, parentElement);
```
Comment (rosensilva) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/navigator/EsbNavigatorContentProvider.java]: why have we deleted these line in this file?

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/deserializer/PropertyMediatorDeserializer.java (hunk 3 above)
Comment (praveennadarajah) [reply plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/deserializer/PropertyMediatorDeserializer.java]: Yes. We are supporting.

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/deserializer/PropertyGroupMediatorDeserializer.java (hunk 2 above)
Comment (praveennadarajah) [reply plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/deserializer/PropertyGroupMediatorDeserializer.java]: Changed

File: plugins/org.wso2.developerstudio.eclipse.esb.libraries/.classpath (hunk 1 above)
Comment (praveennadarajah) [reply plugins/org.wso2.developerstudio.eclipse.esb.libraries/.classpath]: Added new synapse-core version

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/navigator/EsbNavigatorActionProvider.java (hunk 4 above)
Comment (praveennadarajah) [reply plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/navigator/EsbNavigatorActionProvider.java]: Changed to previous code

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/navigator/EsbNavigatorContentProvider.java (hunk 5 above)
Comment (praveennadarajah) [reply plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/navigator/EsbNavigatorContentProvider.java]: No formatting issues

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/navigator/EsbNavigatorContentProvider.java (hunk 6 above)
Comment (praveennadarajah) [reply plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/navigator/EsbNavigatorContentProvider.java]: I have split the large method into two sub methods. So this line is went into the other method

**Pull request merged**